### DISCUSSION TITLE
Is it possible to have only one worker without duration per queue?

### DISCUSSION THREAD

**Discussion by nullndr** [taskforcesh/bullmq#1162] [Q&A] [answered]
I'm implementing a little queue where the jobs MUST be processed one by one, so I thought to use a `limiter`:

```typescript
const worker = new Worker(
  executeBulkOperationQueue.name,
  async(job: Job<ExecuteBulkOperationPayload>) => { ... },
  { 
    connection: redis, 
    limiter: { 
      max: 1,
      duration: 30000, // 30 seconds
    }
  }
)
```

And this works, but my worker has to wait for the remaining seconds even if the job took, for example, 3 seconds.

Is it possible to start the only one worker with the new job once the old job has finished?

**Comment by nullndr** [comment, marked as answer]
Nvm, I forgot about the `concurrency` option and now It works better.

**Reply by manast** [reply]
It will work as long as you only have one worker at all times. Otherwise I think the concurrency feature in groups is exactly what you want, however it is only available in the Pro version: https://docs.bullmq.io/bullmq-pro/groups/concurrency
